src/proto/Harmony.proto

```diff
@@ -45,3 +45,92 @@ message SigningOutput {
     bytes r = 3;
     bytes s = 4;
 }
+
+message StakingTransactionInput {
+    // StakeMsg 
+    oneof stake_msg {
+        DirectiveCreateValidator create_validator_message = 1;
+        DirectiveEditValidator edit_validator_message = 2;
+        DirectiveDelegate delegate_message = 3;
+        DirectiveUndelegate undelegate_message = 4;
+        DirectiveCollectRewards collect_rewards = 5;
+    }
+
+    // Nonce (256-bit number)
+    bytes nonce = 6;
+
+    // Gas price (256-bit number)
+    bytes gas_price = 7;
+
+    // Gas limit (256-bit number)
+    bytes gas_limit = 8;    
+
+    // Chain identifier (256-bit number)
+    bytes chain_id = 9; 
+
+    // Private key.
+    bytes private_key = 10;
+}
+
+message StakingTransactionOutput {
+    // Signed and encoded transaction bytes.
+    bytes encoded = 1;
+
+    bytes v = 2;
+    bytes r = 3;
+    bytes s = 4;
+}
+
+message Description {
+    string name = 1;
+    string identity = 2;
+    string website = 3;
+    string security_contact = 4;
+    string details = 5;
+}
+
+message Decimal {
+    string value = 1;
+}
+
+message CommissionRate {
+    Decimal rate = 1;
+    Decimal max_rate = 2;
+    Decimal max_change_rate = 3;
+}
+
+message DirectiveCreateValidator {
+    string validator_address = 1;
+    Description description = 2;
+    CommissionRate commission_rates = 3;
+    bytes min_self_delegation = 4;
+    bytes max_total_delegation = 5;
+    repeated bytes slot_pub_keys = 6;
+    bytes amount = 7;
+}
+
+message DirectiveEditValidator {
+    string validator_address = 1;
+    Description description = 2;
+    Decimal commission_rate = 3;
+    bytes min_self_delegation = 4;
+    bytes max_total_delegation = 5;
+    bytes slot_key_to_remove = 6;
```

Comment:
Does it differentiate a nil pointer as in our struct "SlotKeyToRemove    *shard.BlsPublicKey `json:"slot_key_to_remove" yaml:"slot_key_to_remove"`"?

Reply:
protobuf handles this case as by default all fields are optional and nilptr is assigned for those unfilled values.